File: julia/src/K256.jl
"""
    K256

Official Julia SDK for K256 - the gateway to Solana's liquidity ecosystem.

Connect any application to Solana's liquidity ecosystem. One API. All venues. Full observability.

# Example

```julia
using K256

# Decode fee market (per-writable-account fees)
fees = K256.decode_fee_market(payload)
println("Recommended fee: \$(fees.recommended) microlamports")

# Decode blockhash
bh = K256.decode_blockhash(payload)
println("Blockhash: \$(bh.blockhash)")

# Base58 encoding
address = K256.base58_encode(pubkey_bytes)
if K256.is_valid_pubkey(address)
    println("Valid pubkey: \$address")
end
```
"""
module K256

export MessageType, NetworkState
export OrderLevel, PoolUpdate, AccountFee, FeeMarket, Blockhash, Quote, Heartbeat, Token
export decode_fee_market, decode_blockhash, decode_pool_update, decode_pool_update_batch, decode_quote
export base58_encode, base58_decode, is_valid_pubkey

const VERSION = "0.1.0"

include("types.jl")
include("base58.jl")
include("decoder.jl")

end # module

File: swift/Sources/K256/K256.swift
/// K256 SDK - The gateway to Solana's liquidity ecosystem
///
/// Official Swift SDK for K256. Connect any application to Solana's
/// liquidity ecosystem. One API. All venues. Full observability.
///
/// This SDK provides binary message decoders and types. For WebSocket
/// client functionality, use a Swift networking library like URLSession
/// or Starscream.
///
/// ## Example
///
/// ```swift
/// import K256SDK
///
/// // Decode fee market from binary payload
/// if let fees = K256Decoder.decodeFeeMarket(payload) {
///     print("Slot: \(fees.slot)")
///     print("Recommended fee: \(fees.recommended) microlamports")
/// }
///
/// // Decode blockhash
/// if let bh = K256Decoder.decodeBlockhash(payload) {
///     print("Blockhash: \(bh.blockhash)")
/// }
///
/// // Base58 encoding
/// let address = Base58.encode(pubkeyBytes)
/// ```

public enum K256 {
    /// SDK version
    public static let version = "0.1.0"
}

// Re-exports
public typealias MessageType = K256MessageType
public typealias NetworkState = K256NetworkState
public typealias OrderLevel = K256OrderLevel
public typealias PoolUpdate = K256PoolUpdate
public typealias AccountFee = K256AccountFee
public typealias FeeMarket = K256FeeMarket
public typealias Blockhash = K256Blockhash
public typealias Quote = K256Quote
public typealias Heartbeat = K256Heartbeat
public typealias Token = K256Token
